xwiki-rendering-macros/xwiki-rendering-macro-message/src/test/resources/macromessage3.test

```diff
@@ -63,6 +63,8 @@ onSpace
 onWord [possibility]
 endListItem
 endList [BULLETED]
+beginMetaData [[unchanged-content]=[true]]
```

Comment:
This does not make much sense to me. It should be around the content, not after it.

Reply:
I agree, it was one of my concern: this is because when doing a message, the content is put from the title parameter, not from its content. So basically it's a box with a title but with an empty content. See: https://github.com/xwiki/xwiki-rendering/blob/225ab59614552a72a5ebf48e332cc0a5ed352abc/xwiki-rendering-macros/xwiki-rendering-macro-message/src/main/java/org/xwiki/rendering/internal/macro/message/AbstractMessageMacro.java#L80-L85

So I wonder how to fix that one properly

Reply:
All I can think of is stop hacking the BoxMacro in AbstractMessageMacro the way we currently do (at least for inline content). It will be even worst when we start generating metadata for the title.

I guess one thing you could do quickly is to check in ContentMacro if the context macro really is a content macro, if it's not then you can't really trust the `content` execute gave you.

Reply:
> stop hacking the BoxMacro in AbstractMessageMacro the way we currently do (at least for inline content)

so you suggest I fix the MessageMacro to properly use the content of the box? 

> you could do quickly is to check in ContentMacro if the context macro really is a content macro

Just checked while executing ContentMacro tests and apparently it is.

Reply:
> so you suggest I fix the MessageMacro to properly use the content of the box?

I suggest to stop using the ContentMacro for inline content or at least not that way (passing the content as title and an empty content) since it's a "lie".

> Just checked while executing ContentMacro tests and apparently it is.

I don't see how since AbstractMessageMacro does not modify the macro context. I'm referring to the following logic: `context.getCurrentMacroBlock().getId().equals("content")`

Reply:
Sorry but I don't follow there: 

> I suggest to stop using the ContentMacro for inline content

I don't see where ContentMacro is used here. It's BoxMacro that is used.

Reply:
Yes box macro sorry, just a typo.

Reply:
ok but maybe this PR is not the better place to do that refactoring properly. Especially since the whole behaviour of MacroMessage is currently relying on Box Macro. 
Do we open a ticket about that?

Reply:
> ok but maybe this PR is not the better place to do that refactoring properly. Especially since the whole behaviour of MacroMessage is currently relying on Box Macro.

That's why I suggested you to add a protection in the macro used by AbstractMessageMacro (which I called the  ContentMacro instead of DefaultBoxMacro) to not end up with meaningless metadata.

Reply:
I opened https://jira.xwiki.org/browse/XRENDERING-524 and https://github.com/xwiki/xwiki-rendering/pull/148 to start working more properly on this issue.
I'll see what I can do for the protection you suggest.

Reply:
I added the protection, but then I had to decrease a bit the coverage (from 96 to 95%): this is because I now have two cases (if I'm in the context of box macro or if I'm not), and both cases are already tested, but the latter is tested inside the macro message now so not counted for jacoco locally in macro box.

Reply:
In box macro module you could add a TestMacro which use the box macro execute directly and use that test macro in a test. That way you would make sure it works but any macro (not just message macro which is going to change anyway).